sdk/Pulumi/Config.cs

```diff
@@ -155,19 +155,28 @@ private static Output<T> MakeStructSecret<T>(T value) where T : struct
             => MakeStructSecret(GetDoubleImpl(key));
 
         [return: MaybeNull]
-        private T GetObjectImpl<T>(string key, string? use = null, [CallerMemberName] string? insteadOf = null)
+        private T GetObjectImpl<T>(string key, string? use = null, [CallerMemberName] string? insteadOf = null, JsonSerializerOptions? options = null)
         {
             var v = GetImpl(key, use, insteadOf);
             try
             {
-                return v == null ? default : JsonSerializer.Deserialize<T>(v);
+                return v == null ? default : JsonSerializer.Deserialize<T>(v, options);
             }
             catch (JsonException ex)
             {
                 throw new ConfigTypeException(FullKey(key), v, typeof(T).FullName!, ex);
             }
         }
 
+        /// <summary>
+        /// Loads an optional configuration value, as an object, by its key, or null if it doesn't
+        /// exist. This works by taking the value associated with <paramref name="key"/> and passing
+        /// it to <see cref="JsonSerializer.Deserialize{TValue}(string, JsonSerializerOptions)"/>.
+        /// </summary>
+        [return: MaybeNull]
+        public T GetObject<T>(string key, JsonSerializerOptions? options)
```

Comment:
GetObeject without the JsonSerializerOptions parameter is already present so it would be better to use a non null able argument here similiar how you did GetSecretObject

Reply:
Fixed